src/ATen/native/xpu/mkl/BatchLinearAlgebra.cpp

```diff
@@ -0,0 +1,1228 @@
+    int64_t batch_size) {
+  return oneapi::mkl::lapack::getrf_batch_scratchpad_size<std::complex<double>>(
+      queue, m, n, lda, stride_a, stride_ipiv, batch_size);
+}
+
+template <>
+int64_t mkl_getrf_scratchpad<c10::complex<float>>(
+    sycl::queue& queue,
+    int64_t m,
+    int64_t n,
+    int64_t lda,
+    int64_t stride_a,
+    int64_t stride_ipiv,
+    int64_t batch_size) {
+  return oneapi::mkl::lapack::getrf_batch_scratchpad_size<std::complex<float>>(
+      queue, m, n, lda, stride_a, stride_ipiv, batch_size);
+}
+
+template <typename scalar_t>
+int64_t mkl_getrs_scratchpad(
+    sycl::queue& queue,
+    oneapi::mkl::transpose trans,
+    int64_t n,
+    int64_t nrhs,
+    int64_t lda,
+    int64_t stride_a,
+    int64_t stride_ipiv,
+    int64_t ldb,
+    int64_t stride_b,
+    int64_t batch_size) {
+  return oneapi::mkl::lapack::getrs_batch_scratchpad_size<scalar_t>(
+      queue,
+      trans,
+      n,
+      nrhs,
+      lda,
+      stride_a,
+      stride_ipiv,
+      ldb,
+      stride_b,
+      batch_size);
+}
+
+template <>
+int64_t mkl_getrs_scratchpad<c10::complex<double>>(
+    sycl::queue& queue,
+    oneapi::mkl::transpose trans,
+    int64_t n,
+    int64_t nrhs,
+    int64_t lda,
+    int64_t stride_a,
+    int64_t stride_ipiv,
+    int64_t ldb,
+    int64_t stride_b,
+    int64_t batch_size) {
+  return oneapi::mkl::lapack::getrs_batch_scratchpad_size<std::complex<double>>(
+      queue,
+      trans,
+      n,
+      nrhs,
+      lda,
+      stride_a,
+      stride_ipiv,
+      ldb,
+      stride_b,
+      batch_size);
+}
+
+template <>
+int64_t mkl_getrs_scratchpad<c10::complex<float>>(
+    sycl::queue& queue,
+    oneapi::mkl::transpose trans,
+    int64_t n,
+    int64_t nrhs,
+    int64_t lda,
+    int64_t stride_a,
+    int64_t stride_ipiv,
+    int64_t ldb,
+    int64_t stride_b,
+    int64_t batch_size) {
+  return oneapi::mkl::lapack::getrs_batch_scratchpad_size<std::complex<float>>(
+      queue,
+      trans,
+      n,
+      nrhs,
+      lda,
+      stride_a,
+      stride_ipiv,
+      ldb,
+      stride_b,
+      batch_size);
+}
+
+template <typename scalar_t>
+int64_t mkl_getri_scratchpad(
+    sycl::queue& queue,
+    int64_t n,
+    int64_t lda,
+    int64_t stride_a,
+    int64_t stride_ipiv,
+    int64_t ldainv,
+    int64_t stride_ainv,
+    int64_t batch_size) {
+  return oneapi::mkl::lapack::getri_batch_scratchpad_size<scalar_t>(
+      queue, n, lda, stride_a, stride_ipiv, ldainv, stride_ainv, batch_size);
+}
+
+template <>
+int64_t mkl_getri_scratchpad<c10::complex<double>>(
+    sycl::queue& queue,
+    int64_t n,
+    int64_t lda,
+    int64_t stride_a,
+    int64_t stride_ipiv,
+    int64_t ldainv,
+    int64_t stride_ainv,
+    int64_t batch_size) {
+  return oneapi::mkl::lapack::getri_batch_scratchpad_size<std::complex<double>>(
+      queue, n, lda, stride_a, stride_ipiv, ldainv, stride_ainv, batch_size);
+}
+
+template <>
+int64_t mkl_getri_scratchpad<c10::complex<float>>(
```

Comment:
The specialization of templates introduces redundancy. We'd better refine it in another PR.